File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(const char* phone_number) {
    int length = strlen(phone_number);
    

    char* answer = (char*)malloc(length + 1);
    
    for (int i = 0; i < length; i++) {
        if (i < length - 4) {
            answer[i] = '*';
        } else {
            answer[i] = phone_number[i];
        }
    }
    
    answer[length] = '\0';
    
    return answer;
}

int main() {
    const char* phone_number = "01012345678";
    char* result = solution(phone_number);
    printf("%s\n", result);
    
    free(result);

    return 0;
}

File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.java
class Solution {
    public String solution(String phone_number) {
        StringBuilder answer = new StringBuilder();
        
        // 전화번호의 앞부분을 '*'로 치환
        for (int i = 0; i < phone_number.length() - 4; i++) {
            answer.append('*');
        }
        
        // 전화번호의 마지막 4자리를 그대로 추가
        answer.append(phone_number.substring(phone_number.length() - 4));

        return answer.toString();
    }
}